File: messaging-frontend/src/types/conversation.ts
import type { Customer } from "./customer"

export type Conversation = {
    id: string,
    owner: Customer,
    lastUpdated: Date
}

export type ConversationResponse = {
    id: string,
    owner: Customer,
    lastUpdated: Date,
    lastMessageSender: string,
    lastMessageContent: string,
    unreadCount: number
}

File: MessagingAndroid/app/src/main/java/ma/tayeb/messaging_android/types/Conversation.kt
package ma.tayeb.messaging_android.types

import org.threeten.bp.LocalDateTime
import java.util.UUID

data class Conversation(
    val id: UUID,
    val owner: Customer,
    val lastUpdated: LocalDateTime
)

data class ConversationResponse(
    val id: UUID,
    val owner: Customer,
    val lastUpdated: LocalDateTime,
    val lastMessageSender: String,
    val lastMessageContent: String,
    val unreadCount: Int
)